onnxruntime/core/optimizer/transpose_optimizer/api.h

```diff
@@ -72,19 +72,35 @@ class TensorRef {
   /// <returns>The dtype of the tensor.</returns>
   virtual DataType DType() const = 0;
 
+  /// <summary>
+  /// Retrieves raw data from the tensor. Used for reading initializers specifying axes/pads/scales.
+  /// </summary>
+  /// <returns>Flattened tensor data in bytes</returns>
+  virtual std::vector<char> Data() const = 0;
+
```

Comment:
What's the benefit of this to justify the reinterpret_cast's and multiple copies of the data (e.g. Tensor -> vector<char> -> vector<intX>)?

Is it mainly so PermuteInput doesn't have a branch on int64 vs int32?